### DISCUSSION TITLE
Support additional keys

### DISCUSSION THREAD

**Discussion by DannyBen** [charmbracelet/vhs#603] [Ideas]
As discussed in #64 - we need the ability to send more keys. Here is a list borrowed from the [AutoHotkey list](https://www.autohotkey.com/docs/commands/Send.htm)

- F1 - F12
- Enter
- Escape + Esc
- Space
- Tab
- Backspace
- Delete + Del
- Insert + Ins
- Up / Down / Left / Right
- End
- PgUp / PgDn
- CapsLock
- ScrollLock
- NumLock
- Control + Ctrl **
- LControl + LCtrl **
- RControl + RCtrl **
- Alt **
- LAlt **
- RAlt **
- Shift **
- LShift **
- RShift **
- LWin **
- RWin **
- AppsKey **
- Numpad0 - Numpad9
- NumpadDot
- NumpadEnter
- NumpadMult
- NumpadDiv
- NumpadAdd
- NumpadSub
- NumpadDel
- NumpadIns
- NumpadClear
- NumpadUp
- NumpadDown
- NumpadLeft
- NumpadRight
- NumpadHome
- NumpadEnd
- NumpadPgUp
- NumpadPgDn
- PrintScreen
- Pause

`** These keys may not be needed in vhs context`

**Comment by Airblader** [comment]
`Escape` seems to be supported already, at least the examples already use it.

**Comment by taminomara** [comment]
As an alternative, maybe it would be a good idea to support sending raw CSI sequences?

**Comment by bashbunni** [comment]
There should be support for a lot of the keys on this list. I'd need to double check the source code to confirm, but you can see the supported keys here:

https://github.com/charmbracelet/vhs/blob/b41d47fa1d2e5847aa3c5fd0e5e2b284fd83f08c/token/token.go#L19-L104

Out of the ones that aren't implemented yet, are there any in particular that users would like to see introduced? If so, please let us know! I'm going to convert this to a discussion in the meantime to track this conversation while keeping our backlog clear :)

**Comment by rkgarcia** [comment]
Function keys! can be nice to have f1...f12
